ldms/src/contrib/store/timescale/store_timescale.c

```diff
@@ -0,0 +1,573 @@
+                        msglog(LDMSD_LERROR, "Invalid password file path! Must start with '/'.\n");
+                        return EINVAL;
+                }
+
+                FILE *file = fopen(pwfile, "r");
+                if (!file) {
+                        msglog(LDMSD_LERROR, "Unable to open password file!\n");
+                        return EINVAL;
+                }
+                char line[600];
+                char *s, *ptr;
+                s = NULL;
+                while (fgets(line, 600, file)) {
+                        if ((line[0] == '#') || (line[0] == '\n'))
+                                continue;
+                        if (0 == strncmp(line, "secretword=", 11)) {
+                                s = strtok_r(&line[11], " \t\n", &ptr);
+                                if (!s) {
+                                        msglog(LDMSD_LERROR, "Auth error: the secret word is an empty srting.\n");
+                                        return EINVAL;
+                                }
+                                break;
+                        }
+                }
+                if (!s) {
+                        msglog(LDMSD_LERROR, "No secret word in the file!\n");
+                        return EINVAL;
+                }
+                strncpy(password, strdup(s), sizeof(password));
+//                if (!password) {
+//                        msglog(LDMSD_LERROR, "Auth error: Out of memory when trying to read the secret word.\n");
+//                        return EINVAL;
+//                }
+                fclose(file);
+        }
+
+        value = av_value(avl, "hostaddr");
+        if (!value) {
+                msglog(LDMSD_LERROR, "The 'hostaddr' keyword is required.\n");
+                return EINVAL;
+        }
+        strncpy(hostaddr, value, sizeof(hostaddr));
+
+        value = av_value(avl, "port");
+        if (!value) {
+                msglog(LDMSD_LERROR, "The 'port' keyword is required.\n");
+                return EINVAL;
+        }
+        strncpy(port, value, sizeof(port));
+
+        value = av_value(avl, "dbname");
+        if (!value) {
+                msglog(LDMSD_LERROR, "The 'dbname' keyword is required.\n");
+                return EINVAL;
+        }
+        strncpy(dbname, value, sizeof(dbname));
+
+	value = av_value(avl, "measurement_limit");
+	if (value) {
+		measurement_limit = strtol(value, NULL, 0);
+		if (measurement_limit <= 0) {
+			msglog(LDMSD_LERROR,
+			       "'%s' is not a valid 'measurement_limit' value\n",
+			       value);
+			measurement_limit = MEASUREMENT_LIMIT_DEFAULT;
+		}
+		return EINVAL;
+	}
+
+	pthread_mutex_unlock(&cfg_lock);
+	return 0;
+}
+
+static void term(struct ldmsd_plugin *self)
+{
+}
+
+static const char *usage(struct ldmsd_plugin *self)
+{
+	return "config name=store_timescale user=<username> pwfile=<full path to password file> hostaddr=<host ip addr> port=<port no> dbname=<database name> measurement_limit=<sql statement length>";
```

Comment:
This is a nit, but in general we try to break up these super-long strings into something like this
```
return "config name=store_timescale user=<username> pwfile=<full path to password file> "
       "hostaddr=<host ip addr> port=<port no> dbname=<database name> "
       "measurement_limit=<sql statement length>";
```
The C compiler automatically combines those strings without the need for the line continuation character '\\'